RHIDP-2405 : Added the content explaining vulnerabilities, SBOMs, and EC (#72)
redhat-developer/red-hat-developers-documentation-rhdh: modules/rhdh-plugins-reference/proc-setting-up-openshift-pipelines-to-view-project-vulnerabilities.adoc

:_mod-docs-content-type: PROCEDURE
[id="setting-up-openshift-pipelines-to-view-project-vulnerabilities_{context}"]

= Setting up OpenShift Pipelines to view project vulnerabilities 

> In Red Hat Developer Hub, the Pipeline Runs section on the *CI* tab page provides a visual representation of identified vulnerabilities,  categorized by the severity (critical, high, medium, and low). This streamlined view facilitates prioritization and remediation efforts.

Reviewer: Consider using the attribute:

```suggestion
In {product}, the Pipeline Runs section on the *CI* tab page provides a visual representation of identified vulnerabilities,  categorized by the severity (critical, high, medium, and low). This streamlined view facilitates prioritization and remediation efforts.
```

> .Viewing vulnerabilities on the `PipelineRun` details page

Reviewer 2: ```suggestion
.Viewing vulnerabilities on the `PipelineRun` list page
```

image::rhdh-plugins-reference/vulnerabilities_details.png[]

[NOTE]
====

> The vulnerability scan task discussed in this chapter uses `roxctl` for example purposes only. `roxctl` is a community project and is not supported by Red Hat Developer Hub by default. For guidance on implementing these or any other community-supported tasks consult your support agreements or contact support services.

Reviewer: Consider using the attribute:

```suggestion
The vulnerability scan task discussed in this chapter uses `roxctl` for example purposes only. `roxctl` is a community project and is not supported by {product} by default. For guidance on implementing these or any other community-supported tasks consult your support agreements or contact support services.
```

====

.Prerequisites

> * You have link:https://docs.openshift.com/container-platform/4.14/web_console/web-console.html#web-console[logged in to the web console].

Reviewer: Consider using the attribute:

```suggestion
* You have link:https://docs.openshift.com/container-platform/4.14/web_console/web-console.html#web-console[logged in to the {ocp-short} web console].
```

> * You have the appropriate link:https://docs.openshift.com/container-platform/4.14/authentication/using-rbac.html#default-roles_using-rbac[roles and permissions] in a project to create applications and other workloads in OpenShift Container Platform.

Reviewer: Consider using the attribute:

```suggestion
* You have the appropriate link:https://docs.openshift.com/container-platform/4.14/authentication/using-rbac.html#default-roles_using-rbac[roles and permissions] in a project to create applications and other workloads in {ocp-short}.
```

          medium: (.result.summary.MODERATE),
          low: (.result.summary.LOW)
            }}' scan_output.json | tee $(results.SCAN_OUTPUT.path)
----
<1> The name of your task.
<2> The location for storing the task outputs.
<3> The naming convention of the scan task result. A valid naming convention must end with the `SCAN_OUTPUT` string. For example, SCAN_OUTPUT, MY_CUSTOM_SCAN_OUTPUT, or ACS_SCAN_OUTPUT.
<4> The description of the result.
<5> The name of the vulnerability scanning tool that you have used. 
<6> The location of the actual image containing the scan tool.
<7> The tool-specific environment variables.
<8> The shell script to be executed with json output. For example, scan_output.json.
<9> The format to extract vulnerability summary (adjust `jq` command for different JSON structures).

. Update an appropriate Pipeline to add vulnerabilities specifications in the following format:

+
[source,yaml]
----
...
spec:
  results:
    - description: The common vulnerabilities and exposures (CVE) result
      name: SCAN_OUTPUT
      type: $(tasks.vulnerability-scan.results.SCAN_OUTPUT)
----

. Rerun the modified OpenShift pipeline.

.Verification

> . Log in to Red Hat Developer Hub.

Reviewer: Consider using the attribute:

```suggestion
. Log in to {product-short}.
```
